### ISSUE TITLE
OpenGlControl resize native crash

### ISSUE THREAD

**Issue by jvbsl** [AvaloniaUI/Avalonia#4425]
In a simple avalonia app only containing an empty OpenGlControl(derived from OpenGlControlBase) crashes after multiple resizes on my Linux Manjaro with Intel GPU.
```
00007FFF3AC03570 00007fff477967f8 [InlinedCallFrame: 00007fff3ac03570] SkiaSharp.SkiaApi.sk_canvas_flush(IntPtr)
00007FFF3AC03570 00007fff7f77c92e [InlinedCallFrame: 00007fff3ac03570] SkiaSharp.SkiaApi.sk_canvas_flush(IntPtr)
00007FFF3AC03560 00007FFF7F77C92E SkiaSharp.SKCanvas.Flush()
00007FFF3AC03600 00007FFF7F77C5A7 Avalonia.Skia.GlRenderTarget+GlGpuSession.Dispose()
00007FFF3AC03620 00007FFF7F77B757 Avalonia.Skia.DrawingContextImpl.Dispose()
00007FFF3AC03660 00007FFF7F7869C9 Avalonia.Rendering.DeferredRenderer.Render(Boolean)
00007FFF3AC03710 00007FFF7EEAB197 Avalonia.Rendering.RenderLoop.TimerTick(System.TimeSpan)
00007FFF3AC037A0 00007FFF7F78669C Avalonia.Rendering.DefaultRenderTimer.InternalTick(System.TimeSpan)
00007FFF3AC037B0 00007FFF7F78638D System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) [/_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs @ 201]
00007FFF3AC03800 00007FFF7F7862D6 System.Threading.TimerQueueTimer.CallCallback(Boolean) [/_/src/System.Private.CoreLib/shared/System/Threading/Timer.cs @ 649]
00007FFF3AC03840 00007FFF7F7860A9 System.Threading.TimerQueueTimer.Fire(Boolean) [/_/src/System.Private.CoreLib/shared/System/Threading/Timer.cs @ 610]
00007FFF3AC03890 00007FFF7F785B27 System.Threading.TimerQueue.FireNextTimers() [/_/src/System.Private.CoreLib/shared/System/Threading/Timer.cs @ 286]
00007FFF3AC03C90 00007ffff7010d1f [DebuggerU2MCatchHandlerFrame: 00007fff3ac03c90]
```

Unfortunately I can't debug further as I'am unable to get SkiaSharp compiled. In the native stacktrace the actual crash seems to happen in the iris_drm.so but even if I switch to the old i965 implementations it still happens, so I don't think it is an actual driver problem. At least not directly, even though it doesn't happen on my nvidia.

One further thing I noticed is that it crashes much sooner when I either resize slowly and continously or if I do it fast and often without much pause, compared to doing it fast but only in intervals(even though the accumulated time is much more of resizing is more than for the other two). That's why I think it could be something threading related, where the first to ways give me a much higher chance to do the resize at an "inconvenient" time, while with the latest method it only happens occasionally if I happen to time it "bad".

I hoped that it perhaps would be fixed by https://github.com/AvaloniaUI/Avalonia/pull/4424 but while the stackframes further down the list change somewhat, the crash itself happens at the exact same stackrames.

Hope someone can help confirm this as well as perhaps debug it further. Thanks in advance.

**Comment by jmacato** [comment]
can you try using #4424 again and dump the native stacktrace?

**Comment by jvbsl** [comment]
```
00007FFF3AC03580 00007fff577947f8 [InlinedCallFrame: 00007fff3ac03580] SkiaSharp.SkiaApi.sk_canvas_flush(IntPtr)
00007FFF3AC03580 00007fff7f79944e [InlinedCallFrame: 00007fff3ac03580] SkiaSharp.SkiaApi.sk_canvas_flush(IntPtr)
00007FFF3AC03570 00007FFF7F79944E SkiaSharp.SKCanvas.Flush()
00007FFF3AC03610 00007FFF7F799337 Avalonia.Skia.GlRenderTarget+GlGpuSession.Dispose()
00007FFF3AC03630 00007FFF7F798957 Avalonia.Skia.DrawingContextImpl.Dispose()
00007FFF3AC03670 00007FFF7F79BF89 Avalonia.Rendering.DeferredRenderer.Render(Boolean)
00007FFF3AC03720 00007FFF7EEBDF96 Avalonia.Rendering.RenderLoop.TimerTick(System.TimeSpan)
00007FFF3AC037C0 00007FFF7EEB74D7 Avalonia.Rendering.SleepLoopRenderTimer.LoopProc()
00007FFF3AC03890 00007FFF7D725362 System.Threading.ThreadHelper.ThreadStart_Context(System.Object) [/_/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs @ 44]
00007FFF3AC038B0 00007FFF7D72EC01 System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) [/_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs @ 172]
00007FFF3AC03900 00007FFF7D72547E System.Threading.ThreadHelper.ThreadStart() [/_/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs @ 93]
00007FFF3AC03C60 00007ffff7010d1f [GCFrame: 00007fff3ac03c60] 
00007FFF3AC03D30 00007ffff7010d1f [DebuggerU2MCatchHandlerFrame: 00007fff3ac03d30]
```

**Comment by jvbsl** [comment]
At the end of the day I don't really know what it was, but it seems to work now(without even updating Avalonia) therefore I guess at some point I got a driver update which fixed it.

**Comment by bmitc** [comment]
@jvbsl Did you ever figure out anything more specific about what was causing this or what solved it? I am using SkiaSharp with my own GLFW bindings, and I am getting a similar error:

```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Repeat 2 times:
--------------------------------
   at SkiaSharp.SkiaApi.sk_canvas_flush(IntPtr)
--------------------------------
   at SkiaSharp.SKCanvas.Flush()
   at Windowing.Window+Window.Draw()
   at <StartupCode$WindowTest>.$Program.main@()
```

I'm stuck and can't figure out what's going on, and this is the only other mention of this error according to Google. :) What I am drawing is very simple, which is just a canvas clear for the background and then a single circle. I do not get the error when just clearing the canvas (and leaving off the circle).

See https://github.com/mono/SkiaSharp/issues/2125

**Comment by MeltyPlayer** [comment]
It looks like this issue is still occurring? Resizing the window consistently crashes for me.

In https://github.com/AvaloniaUI/Avalonia/discussions/13239, folks say that this issue (or one that was similar) occurred in Avalonia 11.0.6 and was fixed since then, but I'm currently on 11.0.10. Maybe there was a regression or the root issue was never really fixed?

**Comment by MeltyPlayer** [comment]
Following up on this, I'm not seeing this issue anymore after updating to the newest version. Guess it's been fixed internally or upstream?
